cpp/Platform.Converters/CachingConverterDecorator.h

```diff
@@ -1,15 +1,32 @@
-﻿namespace Platform::Converters
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <unordered_map>
+
+namespace Platform::Converters
 {
-    template <typename ...> class CachingConverterDecorator;
-    template <typename TSource, typename TTarget> class CachingConverterDecorator<TSource, TTarget> : public IConverter<TSource, TTarget>
+    template <typename ...> class Cached;
+    template <typename TSource, typename TTarget> class Cached<TSource, TTarget>
```

Comment:
@Konard What do you think about changing name?

Reply:
I asked to change it.